backend/src/controllers/encounter.controller.ts

```diff
@@ -2,40 +2,79 @@
  * Controller contains high-level operations using services, consumed by routes
  */
  import { NextFunction, Request, Response } from 'express';
-
- import EncounterModel from '../models/encounter.model';
  import encounterService from '../services/encounter.service';
  import logger from '../utils/logger';
+import httpStatus from "http-status";
+import {EncounterModel} from "../models/encounter.model";
+import {getUserByAuthId} from "../services/user.service";
  
  export const createEncounter = async (
    req: Request,
    res: Response,
    next: NextFunction,
  ): Promise<void> => {
-   logger.info("POST /encounter/create request from frontend");
+   logger.info("POST /encounter request from frontend");
  
    try {
      // Grab the data from the req
      const encounterReq = getEncounterFromReqBody(req.body);
- 
+
      // Pass data to service and attempt to save
      const createdEncounter = await encounterService.createEncounter(encounterReq);
- 
      // Notify frontend that the operation was successful
-     res.sendStatus(200);
+     res.status(httpStatus.CREATED).json(createdEncounter).end();
    } catch (e) {
      next(e);
    }
  };
+export const updateEncounter = async (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+): Promise<any> => {
+    logger.info("PUT /encounter/:id request from frontend");
+
+    try {
+        // check authorization is not a null object (this is necessary check for lint)
+        if (req.headers.authorization == null)
+            return res.status(httpStatus.NOT_FOUND).end()
+
+        const encounterIdToUpdate = req.params.id
+        const firebaseAuthId = req.headers.authorization['user_id']
+
+        // check that user exists with that firebase auth_id, and the user specified encounter actually exists
+        const userByAuthId = await getUserByAuthId(firebaseAuthId);
+        if(!userByAuthId || !isEncounterExistsInUser(userByAuthId, encounterIdToUpdate))
+            return res.status(httpStatus.NOT_FOUND).end()
+
+        // update encounter
+        const newEncounterData = getEncounterFromReqBody(req.body);
+        const updatedEncounter = await encounterService.updateEncounter(encounterIdToUpdate, newEncounterData);
+        return res.sendStatus(updatedEncounter?httpStatus.NO_CONTENT:httpStatus.NOT_FOUND).end();
```

Comment:
If we update the encounter should we return its json with an okay response? I think that's generally what the approach is.

Reply:
Thanks for your comment, Adam. As per [this answer](https://stackoverflow.com/a/827045), PUT should not return a response body unless a resource is created (which isn't the case for us, of course).

> HTTP status code 200 OK for a successful PUT of an update to an existing resource. No response body needed. (Per [Section 9.6](http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.6), 204 No Content is even more appropriate.)

The original [API design](https://docs.google.com/document/d/1MeariGED4lGISSpuRZGJEPHLWrbdB-767JPwCgzo4us/edit#heading=h.d8isee3scc04) is consistent with this too. Since frontend will be able to refer to it through GET /api/encounters/:id, there probably isn't a use case where PUT has to return json.

Reply:
Ah okay good to know. Thanks!